src/dht/dht-peer-routing.js

```diff
@@ -26,14 +27,53 @@ class DHTPeerRouting {
    * @param {any} options
    */
   async findPeer (peerId, options = {}) {
```

Comment:
I think this function should just wait for the `FINAL_PEER` event, then load the multiaddrs from the address book in the peer store and return a `PeerData` object made from those.

The gotcha is that go-kad-dht emits `FINAL_PEER` events without multiaddrs which is why this function looks for `PEER_RESPONSE` events instead in an effort to follow the same logic.

js-kad-dht emits `peer` events [when it discovers closer or provider peers](https://github.com/libp2p/js-libp2p-kad-dht/blob/master/src/network.js#L190-L196) - those events are used to populate the address book, so if we listen for `FINAL_PEER`, then use the address book to load the multiaddrs we shouldn't need to juggle both sets of multiaddrs here.

Reply:
>  then load the multiaddrs from the address book in the peer store and return a PeerData object made from those.

I rather suggest to only return those addresses that the DHT request brought us to be able to get the difference.

> The gotcha is that go-kad-dht emits FINAL_PEER events without multiaddrs which is why this function looks for PEER_RESPONSE events instead in an effort to follow the same logic.

Ok, makes sense. `js-kad-dht` definitely seems to emit FINAL_PEER events *with* multiaddrs. And in these cases, the DHT were able to find addresses but `findPeer` still returns with a `NOT_FOUND` error. This is the main reason led me create the PR.